app/src/main/java/com/android/shelfLife/ui/newoverview/EditFoodItem.kt

```diff
@@ -45,118 +45,124 @@ fun EditFoodItemScreen(
   val context = LocalContext.current
   val coroutineScope = rememberCoroutineScope()
 
-  Scaffold(
-      modifier = Modifier.fillMaxSize(),
-      topBar = {
-        CustomTopAppBar(
-            onClick = { navigationActions.goBack() },
-            title = stringResource(id = R.string.edit_food_item_title),
-            titleTestTag = "editFoodItemTitle",
-            actions = {
-              IconButton(
-                  onClick = {
-                    coroutineScope.launch {
-                      foodItemViewModel.deleteFoodItem()
-                      navigationActions.navigateTo(Route.OVERVIEW)
-                    }
-                  },
-                  modifier = Modifier.testTag("deleteFoodItem")) {
-                    Icon(imageVector = Icons.Default.Delete, contentDescription = "Delete Icon")
-                  }
-            })
-      }) { innerPadding ->
-        LazyColumn(
-            modifier =
-                Modifier.fillMaxSize()
-                    .padding(paddingValues)
-                    .padding(innerPadding)
-                    .testTag("editFoodItemScreen"),
-            horizontalAlignment = Alignment.CenterHorizontally,
-            verticalArrangement = Arrangement.Top) {
-              item(key = "amountAndUnit") {
-                Row(
-                    modifier = Modifier.fillMaxWidth(),
-                    verticalAlignment = Alignment.CenterVertically) {
-                      AmountField(
-                          amount = foodItemViewModel.amount,
-                          onAmountChange = { newValue -> foodItemViewModel.changeAmount(newValue) },
-                          amountErrorResId = foodItemViewModel.amountErrorResId,
-                          modifier = Modifier.weight(1f),
-                          testTag = "editFoodAmount")
-                      Spacer(modifier = Modifier.width(8.dp))
-                      // Display unit as card (non-editable)
-                      Card(
-                          border = CardDefaults.outlinedCardBorder(),
-                          shape = MaterialTheme.shapes.large,
-                          modifier = Modifier.weight(1f).testTag("editFoodUnit")) {
-                            Text(
-                                text = foodItemViewModel.unit.name,
-                                modifier = Modifier.padding(12.dp))
-                          }
+  if (foodItemViewModel.selectedFood != null) {
```

Comment:
## Simplify the entire file changes
Funnily enough, this change made git think you removed and added all the lines in the component
This following code, that you could stick at the top could maybe avoid this by navigating to the easter egg instead of rendering the entire scaffold under a condition.

```suggestion
  if (foodItemViewModel.selectedFood == null) {
  navigationActions.navigateTo(Screen.EASTER_EGG)
  }
```

Reply:
also applicable to [IndividualFoodItem.kt](https://github.com/commit-crimes/shelf-life/pull/303/files#diff-93ab6fb7a5b08067c596dde6d08647bc39c6a348ae3fa5f1291974156d33db5b)